assets/js/app.js

```diff
@@ -0,0 +1,288 @@
+            coinElement.innerText = `Selected: ${coinId}`;
+            selectedCoinsContainer.appendChild(coinElement);
+        });
+    };
+
+    // Display currency information in the modal
+    const showCoinInfo = (coin) => {
+        const modalHtml = `
+            <div class="modal fade" id="coinModal" tabindex="-1" aria-hidden="true">
+                <div class="modal-dialog">
+                    <div class="modal-content">
+                        <div class="modal-header">
+                            <h5 class="modal-title">${coin.name} (${coin.symbol.toUpperCase()})</h5>
+                            <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+                        </div>
+                        <div class="modal-body">
+                            <img src="${coin.image?.large || ''}" alt="${coin.name}" class="img-fluid mb-3">
+                            <p>Current Price:</p>
+                            <ul>
+                                <li>USD: ${coin.market_data?.current_price?.usd ? `$${coin.market_data.current_price.usd}` : 'N/A'}</li>
+                                <li>EUR: ${coin.market_data?.current_price?.eur ? `€${coin.market_data.current_price.eur}` : 'N/A'}</li>
+                                <li>ILS: ${coin.market_data?.current_price?.ils ? `₪${coin.market_data.current_price.ils}` : 'N/A'}</li>
+                            </ul>
+                        </div>
+                    </div>
+                </div>
+            </div>`;
+        document.body.insertAdjacentHTML("beforeend", modalHtml);
+        const modal = new bootstrap.Modal(document.getElementById("coinModal"));
+        modal.show();
+
+        document.getElementById("coinModal").addEventListener("hidden.bs.modal", () => {
+            document.getElementById("coinModal").remove();
+        });
+    };
+
+    // Saving the selected coins in localStorage
+    const saveSelectedCoins = () => {
+        const selectedCoins = Array.from(document.querySelectorAll('.coin-switch:checked')).map(switchElement => switchElement.id.split('-')[1]);
+        localStorage.setItem('selectedCoins', JSON.stringify(selectedCoins));
+    };
+
+    // Loading the selected coins from localStorage
+    const loadSelectedCoins = () => {
+        const savedCoins = localStorage.getItem('selectedCoins');
+        return savedCoins ? JSON.parse(savedCoins) : [];
+    };
+
+    // Create the graph
+    let chart;
+    let dataPoints = {};
+
+    const initializeChart = () => {
+        chart = new CanvasJS.Chart("chartContainer", {
+            title: {
+                text: "Cryptocurrency Price Chart"
+            },
+            axisX: {
+                title: "Time"
+            },
+            axisY: {
+                title: "Price (USD)"
+            },
+            data: [],
+        });
+        chart.render();
+    };
+
+    // Update the graph
+    const updateChart = (data) => {
```

Comment:
the chart display is broken
<img width="1017" alt="Screen Shot 2025-01-29 at 13 30 19" src="https://github.com/user-attachments/assets/e5be1759-ec6b-44c1-81bf-855787cb1ce8" />